test/templates/realtime.tpl.cpp

```diff
@@ -24,7 +36,20 @@ int main() {
     return 1;
   }
 
+  std::streambuf *cout_original = std::cout.rdbuf();
+  std::ostringstream cout_copy;
+
+  std::cout.rdbuf(cout_copy.rdbuf());
+
   [snippet-code]
-  sleep(1);
+
+  while (is_empty(&cout_copy)) {
```

Comment:
You need to break from the loop after a maximum number of retries, to prevent infinite loops if the test fails